imageio/plugins/spe.py

```diff
@@ -707,26 +708,32 @@ def _get_data(self, index):
             if index >= self._len:
                 raise IndexError("Image index %i > %i" % (index, self._len))
 
+            frame_size = self._shape[0] * self._shape[1]
+            meta = self._get_meta_data(index)
+            try:
+                et = ET.fromstring(meta.get("__xml", ""))
+                (stride,) = {
+                    int(e.attrib["stride"])
+                    for e in et.findall(".//{*}DataBlock[@type='Frame'][@stride]")
+                }
```

Comment:
We are creating a set here, which we immediately unpack; any reason why we want to combine stride values that occur multiple times?

If I understand correctly, we expect exactly one DataBlock here (frame-level metadata), if my understanding is correct, wouldn't a simple `et.find(...)` be cleaner, since it avoids any unwrapping/filtering/etc.?

Reply:
At least in theory I guess(?) you can have multiple DataBlocks with different strides, which we simply don't support (I guess in this specific case we should indeed raise, because the contiguous-size assumption definitely doesn't work).